src/SFML/Window/Unix/WindowImplX11.cpp

```diff
@@ -871,7 +875,14 @@ void WindowImplX11::setSize(const Vector2u& size)
         XFree(sizeHints);
     }
 
-    XResizeWindow(m_display, m_window, size.x, size.y);
+    m_resizeOccuredDuringSetSize = false;
+    sf::Clock loopStartTimer;
+    while((getSize() != size) &&
+        !m_resizeOccuredDuringSetSize &&
+        (loopStartTimer.getElapsedTime() < sf::milliseconds(50))) {
```

Comment:
Looks like SDL is doing 100ms (not 10ms as you said originally), I'd also go with 100ms then 😄